lib/persister/game_data.rb

```diff
@@ -1,10 +1,26 @@
 require_relative '../game_mode/lite_3';
 
-class GameData
-    def initialize(game)
+module GameData
+    extend self
+
+    def serializer_types(game)
         @game = game
+        types.map do |type|
```

Comment:
Could you use `yield` to reduce duping in deleting certain keys from the types

Reply:
Ended up using a reduce fn instead to pick out my desired keys. Thought it would be clearer than trying to delete a key, and its immutable

```
    def serializer_types(game)
        @game = game
        types.reduce([]) do |array, type|
            array.push({ 
                    name: type.fetch(:name), 
                    data: type.fetch(:data) 
                })
            array
        end
    end

    def deserializer_types
        types.reduce([]) do |array, type|
            if type.fetch(:deserialize, false)
                array.push({ 
                    name: type.fetch(:name), 
                    deserialize: type.fetch(:deserialize) 
                })
            end
            array
        end
    end
```